crates/swc_ecma_minifier/tests/terser/compress/issue_597/beautify_off_1/output.js

```diff
@@ -1,2 +1,2 @@
-var NaN1;
+var NaN;
 console.log(null, void 0, 1 / 0, NaN, NaN, (1 / 0).toString(), NaN.toString(), NaN.toString());
```

Comment:
The input here is
```js
console.log(null, undefined, Infinity, NaN, Infinity * undefined, ...)
)
```
The 4th and 5th arg should be different.

Reply:
crates/swc_ecma_minifier/tests/terser/compress/issue_597/beautify_off_1/output.terser.js

This is another issue. When you examine the output of Terser, you will discover that Terser avoids pointing to the global NaN when it encounters user-defined NaN.

```JavaScript
var NaN;
console.log(
    null,
    void 0,
    1 / 0,
    NaN,
    0 / 0,
    (1 / 0).toString(),
    NaN.toString(),
    (0 / 0).toString()
);
```

crates/swc_ecma_minifier/tests/terser/compress/issue_597/NaN_and_Infinity_should_not_be_replaced_when_they_are_redefined_evaluate

The intention can be inferred from the name of this test.

Reply:
That's in script mode.